src/libcore/cmp.rs

```diff
@@ -289,6 +384,15 @@ pub fn max<T: Ord>(v1: T, v2: T) -> T {
 /// Compare and return the minimum of two values if there is one.
 ///
 /// Returns the first argument if the comparison determines them to be equal.
+///
+/// # Examples
+///
+/// ```
+/// use std::cmp;
+///
+/// assert_eq!(Some(1), cmp::partial_min(1, 2));
+/// assert_eq!(Some(2), cmp::partial_min(2, 2));
```

Comment:
Similarly these could demonstrate returning `None`, also using NaN.

(Totally minor: Is there a reason that these ones have `cmp::partial_min(2, 2)` but the non-partial ones don't?)